src/shell.js

```diff
@@ -149,7 +149,9 @@ var ENVIRONMENT_IS_WASM_WORKER = Module['$ww'];
 var _scriptDir = (typeof document != 'undefined' && document.currentScript) ? document.currentScript.src : undefined;
 
 if (ENVIRONMENT_IS_WORKER) {
-  _scriptDir = self.location.href;
+  if (typeof self.location == 'object') {
+    _scriptDir = self.location.href;
+  }
```

Comment:
If we don't set `_scriptDir` here, what does it end up being set to in cloudflare workers?    Is it set to something above or is it `undefined`?

What effect does this have on the resulting program?  Presumably its needed?

Reply:
You upload all Wasms at deploy time, then Cloudflare makes them available at static import time, handing you back a [`WebAssembly.Module`](https://developer.mozilla.org/en-US/docs/WebAssembly/JavaScript_interface/Module).

Which can then be used in a `Module.instantiateWasm` callback.